src/imp.rs

```diff
@@ -264,6 +265,66 @@ where
         Ok((inputs, outputs))
     }
 
+    /// Execute a generic contract call
+    #[allow(clippy::too_many_arguments)]
+    pub fn execute<Rng, C>(
+        &self,
+        rng: &mut Rng,
+        contract_id: ContractId,
+        call_data: C,
+        sender_index: u64,
+        refund: &PublicSpendKey,
+        gas_limit: u64,
+        gas_price: u64,
+    ) -> Result<Transaction, Error<S, SC, PC>>
+    where
+        Rng: RngCore + CryptoRng,
+        C: Canon,
+    {
+        let sender = self
+            .store
+            .retrieve_ssk(sender_index)
+            .map_err(Error::from_store_err)?;
+
+        let (inputs, outputs) = self.inputs_and_change_output(
+            rng,
+            &sender,
+            refund,
+            gas_limit * gas_price,
+        )?;
+
+        let blinder = JubJubScalar::random(rng);
+
+        // Since we're not transferring value *to* the contract the crossover
+        // shouldn't contain a value. As such the note used to create it should
+        // be valueless as well.
+        let note = Note::obfuscated(rng, refund, 0, blinder);
+        let (mut fee, crossover) = note
+            .try_into()
+            .expect("Obfuscated notes should always yield crossovers");
+
+        fee.gas_limit = gas_limit;
+        fee.gas_price = gas_price;
```

Comment:
Since we're not transferring to Dusk to or from a contract, the crossover should be `None`. This should look more like `transfer` does:

https://github.com/dusk-network/wallet-core/blob/310efc5040c3417577bb19845e96125966145d7d/src/imp.rs#L297-L298

Reply:
Thanks for the catch, I removed it.